src/distance.cpp

```diff
@@ -0,0 +1,1528 @@
+//#define indexTraverse
+//#define printDistances
+
+#include "distance.hpp"
+
+using namespace std;
+namespace vg {
+
+int64_t minPos (vector<int64_t> vals) {
+//TODO: Move this somewhere else
```

Comment:
Having a helper function in a .cpp file like this is a little unsafe. It won't interfere with code anywhere else, because other files are only aware of the functions in the .hpp file. However, you do run the risk that someone else will define a function with the same name in their .hpp file and clash with yours here.

I'd recommend moving this into a method of whichever class is using this function. If it's used by multiple classes, you could make it a public static method. Static methods don't require you to actually instantiate the class to use the method, but then they are also not allowed to access the member variables of the class (since it may or may not exist). Essentially, they're a way of using the class as a namespace for functions like this. It's organizational more than anything else.